### PR TITLE
feat: introduce AvatarRealm for Plugin

### PR DESCRIPTION
**Pull request by UncleBill**
https://mask.atlassian.net/browse/MF-1479

### CODE REVIEW

File: packages/mask/src/components/InjectedComponents/Avatar.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+export function Avatar(props: AvatarProps) {
+    const { identity } = props
+    const classes = useStylesExtends(useStyles(), props)
+
+    const { value: socialAddressList = EMPTY_LIST, loading: loadingSocialAddressList } =
+        useSocialAddressListAll(identity)
+
+    const component = useMemo(() => {
+        const Component = createInjectHooksRenderer(useActivatedPluginsSNSAdaptor.visibility.useAnyMode, (plugin) => {
+            const shouldDisplay = plugin.AvatarRealm?.Utils?.shouldDisplay?.(identity, socialAddressList) ?? true
```
Comment (guanbinrui) [inline packages/mask/src/components/InjectedComponents/Avatar.tsx]:
```suggestion
            const shouldDisplay = plugin.AvatarRealm?.Utils?.shouldDisplay?.(identity, socialAddressList, sourceType) ?? true
```

File: packages/mask/src/social-network-adaptor/twitter.com/utils/selector.ts [hunk 2]
```diff
@@ -314,6 +314,14 @@ export const searchTwitterAvatarNFTSelector = () =>
 
 export const searchTwitterAvatarNFTLinkSelector = () => querySelector<E>('a[href$="/nft"]')
 
+export const inpageAvatarSelector = () =>
+    querySelectorAll<HTMLDivElement>(
+        [
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/twitter.com/utils/selector.ts:319]: Please leave a comment for each selector.

File: packages/mask/src/components/InjectedComponents/Avatar.tsx [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+    const component = useMemo(() => {
+        const Component = createInjectHooksRenderer(useActivatedPluginsSNSAdaptor.visibility.useAnyMode, (plugin) => {
+            const shouldDisplay =
+                plugin.AvatarRealm?.Utils?.shouldDisplay?.(identity, socialAddressList, sourceType) ?? true
+            return shouldDisplay ? plugin.AvatarRealm?.UI?.Decorator : undefined
+        })
+
+        return <Component identity={identity} />
+    }, [])
```
Comment (nuanyang233) [inline packages/mask/src/components/InjectedComponents/Avatar.tsx]: If `identity` has been changed, the component will not change.

**Review by guanbinrui — APPROVED** [review]

File: packages/mask/src/components/InjectedComponents/Avatar.tsx [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+export interface AvatarProps extends withClasses<'root'> {
+    identity: SocialIdentity
+    sourceType?: Plugin.SNSAdaptor.AvatarRealmSourceType
+}
+
+export function Avatar(props: AvatarProps) {
+    const { identity, sourceType } = props
+    const classes = useStylesExtends(useStyles(), props)
+
```
Comment (nuanyang233) [inline packages/mask/src/components/InjectedComponents/Avatar.tsx:21]: `const { classes } = useStyle(undefined, { props: { classes: props.classes } })`

**Pull request merged**